src/gui/rss/automatedrssdownloader.cpp

```diff
@@ -384,31 +384,51 @@ void AutomatedRssDownloader::on_browseSP_clicked()
 
 void AutomatedRssDownloader::on_exportBtn_clicked()
 {
-//    if (m_editableRuleList->isEmpty()) {
-//        QMessageBox::warning(this, tr("Invalid action"), tr("The list is empty, there is nothing to export."));
-//        return;
-//    }
-//    // Ask for a save path
-//    QString save_path = QFileDialog::getSaveFileName(this, tr("Where would you like to save the list?"), QDir::homePath(), tr("Rules list (*.rssrules)"));
-//    if (save_path.isEmpty()) return;
-//    if (!save_path.endsWith(".rssrules", Qt::CaseInsensitive))
-//        save_path += ".rssrules";
-//    if (!m_editableRuleList->serialize(save_path)) {
-//        QMessageBox::warning(this, tr("I/O Error"), tr("Failed to create the destination file"));
-//        return;
-//    }
+    if (RSS::AutoDownloader::instance()->rules().isEmpty()) {
+        QMessageBox::warning(this, tr("Invalid action")
+                             , tr("The list is empty, there is nothing to export."));
+        return;
+    }
+
+    QString path = QFileDialog::getSaveFileName(
+                this, tr("Where would you like to save the list?")
+                , QDir::homePath(), tr("Rules list (legacy)") + QString(" (*.rssrules)"));
```

Comment:
Why legacy? Is there a new format available or you going to add it in the future?

Reply:
>you going to add it in the future?

Yes. I don't like binary format for it.